### PR TITLE
Run roop using docker-compose with Git clone

### PR DESCRIPTION
**Pull request by janettefuchs**
I tried to address all the issues from https://github.com/s0md3v/roop/pull/404
This pull request should be as close to your requirements as possible.

It doesn't work though, because `requirements.txt` needs to be fixed first:
```
INFO: pip is looking at multiple versions of onnxruntime-gpu to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 14), -r requirements.txt (line 4), -r requirements.txt (line 5), -r requirements.txt (line 6) and numpy==1.23.5 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested numpy==1.23.5
    opencv-python 4.7.0.72 depends on numpy>=1.21.2; python_version >= "3.10"
    opencv-python 4.7.0.72 depends on numpy>=1.22.0; python_version >= "3.11"
    opencv-python 4.7.0.72 depends on numpy>=1.17.0; python_version >= "3.7"
    opencv-python 4.7.0.72 depends on numpy>=1.17.3; python_version >= "3.8"
    opencv-python 4.7.0.72 depends on numpy>=1.19.3; python_version >= "3.9"
    onnx 1.14.0 depends on numpy
    insightface 0.7.3 depends on numpy
    onnxruntime-gpu 1.15.0 depends on numpy>=1.24.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
The command '/bin/sh -c cd /app && pip install -r requirements.txt' returned a non-zero code: 1
ERROR: Service 'roop' failed to build : Build failed
```

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM python:3.11
+
+RUN apt-get update && apt-get install -y \
+	ffmpeg
+
+RUN git clone https://github.com/s0md3v/roop.git /app
```
Comment (henryruhs) [inline docker/Dockerfile:6]: `/root` instead of `/app`

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+RUN cd /app && pip install -r requirements.txt
```
Comment (henryruhs) [inline docker/Dockerfile:8]: RUN pip install -r roop/requirements.txt

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+ENTRYPOINT ["python", "/app/run.py"]
```
Comment (henryruhs) [inline docker/Dockerfile:10]: ENTRYPOINT ['python', '/roop/run.py']

File: docker/Dockerfile.cuda [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y \
+	ffmpeg \
+	nvidia-cudnn \
+	python3 \
+	python3-pip \
+	python3-tk \
+	python-is-python3
+
+WORKDIR /app
```
Comment (henryruhs) [inline docker/Dockerfile.cuda:12]: Same like the other Dockerfile - should be a production only container

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    build:
+      context: ..
+      dockerfile: docker/Dockerfile.cuda
+    deploy:
+     resources:
+      reservations:
+        devices:
+          - driver: nvidia
+            count: 1
+            capabilities: [gpu,video]
```
Comment (henryruhs) [inline docker/docker-compose.yml:21]: gpu, video

**Comment by henryruhs** [comment]
@janettefuchs According to the unresolved dependencies: I have no issues using Python 3.9 in GitHub Actions on Ubuntu 22

File: docker/Dockerfile (hunk 1 above)
Comment (janettefuchs) [reply docker/Dockerfile]: fixed (I hope you mean `/roop` as you do in other comments)

File: docker/Dockerfile (hunk 2 above)
Comment (janettefuchs) [reply docker/Dockerfile]: fixed

File: docker/Dockerfile (hunk 3 above)
Comment (janettefuchs) [reply docker/Dockerfile]: fixed

File: docker/Dockerfile.cuda (hunk 4 above)
Comment (janettefuchs) [reply docker/Dockerfile.cuda]: fixed

File: docker/docker-compose.yml (hunk 5 above)
Comment (janettefuchs) [reply docker/docker-compose.yml]: fixed

**Comment by janettefuchs** [comment]
Usage:

1. Build it:
```
cd docker
docker-compose build roop
```

2. Run it (from git root for the examples to be mounted correctly):
```
docker run \
    --rm \
    -v pwd/.github/examples:/data \
    docker_roop \
    -f '/data/face.jpg' \
    -t '/data/target.mp4' \
    -o '/data/demo.mp4'
```
It fails as of now, but will work as soon as next is merged into main: File "inswapper_128.onnx" does not exist!

**Pull request closed without merging**